riscv/interactive.cc

```diff
@@ -179,7 +179,13 @@ reg_t sim_t::get_pc(const std::vector<std::string>& args)
 
 void sim_t::interactive_pc(const std::string& cmd, const std::vector<std::string>& args)
 {
-  fprintf(stderr, "0x%016" PRIx64 "\n", get_pc(args));
+  if(args.size() != 1)
+    throw trap_interactive();
+
+  processor_t *p = get_core(args[0]);
+  int max_xlen = p->get_max_xlen();
+  fprintf(stderr, max_xlen==32 ? "0x%08" PRIx32 "\n" :
```

Comment:
This should be `PRIx64` even for RV32, since `get_pc` returns a `reg_t` a.k.a. `uint64_t` regardless of `max_xlen`.

Reply:
Changed all `PRIx32` I added to `PRIx64`.

Reverted the definition of `TARGET_ARCH` back to what it was because the problem can be solved by adding `CXXFLAGS=\\\"-DTARGET_ARCH=riscv32-unknown-elf\\\""` to the configure command.

Now I am investigating other bugs that have shown up in spike 32-bit. They are related to memory.

Reply:
All detected bugs corrected.

Reply:
There is a mistake in my last comment:.

he correct addition to the configure command is `CXXFLAGS="-DTARGET_ARCH=\\\"riscv32-unknown-elf\\\""`